### ISSUE TITLE
Authenticator client sometimes panics when trying to kick off authentication

### ISSUE THREAD

**Issue by izgeri** [cyberark/conjur-authn-k8s-client#252]
## Summary
The v0.19.1 authenticator client panics and fails for some applications. The customer reported that most applications were up and running fine, but one of them had this issue:
```
INFO: main.go:16: CAKC048 Kubernetes Authenticator Client v0.19.1-ee98afe starting up...
INFO: authenticator.go:207: CAKC040 Authenticating as user 'host/conjur/authn-k8s/ut-dev/apps/oa/stateful_set/hazelcast-node-analytic-data'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x7c2751]
```

> These ... are running hazelcast (https://hazelcast.com/) which makes me suspicious of something related to that app, but then again, it’s been working fine for months on 0.16.0. The deployments/pods have no memory limits set. I have tried a few things to troubleshoot including deployments with some versions of the authenticator between 0.16.0 and 0.19.1. This produced some interesting results:
> 
> - 0.16.0, 0.16.1: Works great without a problem
> - 0.18.1, 1.19.0, 1.19.1: All of these versions fail with the above errors.

## Steps to Reproduce
TBD - so far we have not been able to reproduce this locally

## Expected Results
The authenticator client delivers an access token as expected.

## Actual Results (including error logs, if applicable)
The authenticator client panics and fails.

More detailed logs:
```
NFO:  2021/03/17 19:55:01.166872 main.go:16: CAKC048 Kubernetes Authenticator Client v0.19.1-ee98afe starting up...
INFO:  2021/03/17 19:55:01.253013 authenticator.go:207: CAKC040 Authenticating as user 'host/conjur/authn-k8s/ut-dev/apps/oa/stateful_set/hazelcast-node-analytic-data'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x7c2751]

goroutine 1 [running]:
github.com/cyberark/conjur-authn-k8s-client/pkg/utils.VerifyFileExists(0x84dc32, 0x1a, 0x0, 0xc00043e000)
	/opt/conjur-authn-k8s-client/pkg/utils/file.go:50 +0xb1
github.com/cyberark/conjur-authn-k8s-client/pkg/utils.WaitForFile.func1(0xc00009a060, 0xc000092080)
	/opt/conjur-authn-k8s-client/pkg/utils/file.go:38 +0x46
github.com/cenkalti/backoff.RetryNotify(0xc000407a38, 0x8a7e00, 0xc00009a060, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37 +0x210
github.com/cenkalti/backoff.Retry(...)
	/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24
github.com/cyberark/conjur-authn-k8s-client/pkg/utils.WaitForFile(0x84dc32, 0x1a, 0xa, 0x860958, 0x0, 0x2d5)
	/opt/conjur-authn-k8s-client/pkg/utils/file.go:33 +0xfb
github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator.(*Authenticator).Login(0xc000428000, 0xc0003cf500, 0x6f)
	/opt/conjur-authn-k8s-client/pkg/authenticator/authenticator.go:141 +0x325
github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator.(*Authenticator).loginIfNeeded(0xc000428000, 0x847322, 0x4)
	/opt/conjur-authn-k8s-client/pkg/authenticator/authenticator.go:239 +0x15c
github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator.(*Authenticator).Authenticate(0xc000428000, 0xc000407c98, 0x61f1b3)
	/opt/conjur-authn-k8s-client/pkg/authenticator/authenticator.go:209 +0xaa
main.main.func1(0xc00042a000, 0xc00028d9e0)
	/opt/conjur-authn-k8s-client/cmd/authenticator/main.go:41 +0x45
github.com/cenkalti/backoff.RetryNotify(0xc000407ea0, 0x8a7d80, 0xc00042a000, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37 +0x210
github.com/cenkalti/backoff.Retry(...)
	/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24
main.main()
	/opt/conjur-authn-k8s-client/cmd/authenticator/main.go:39 +0x206
```

## Reproducible
   * [ ] Always 
   * [x] Sometimes
   * [ ] Non-Reproducible
   
## Version/Tag number
v0.19.1

## Environment setup
Unknown

## Additional Information
n/a

**Comment by izgeri** [comment]
Note: we were able to reproduce a panic in this short example: https://play.golang.org/p/rhFrJ9ZAnkt

**Comment by izgeri** [comment]
This issue was introduced in [v0.18.1](https://github.com/cyberark/conjur-authn-k8s-client/releases/tag/v0.18.1) in #119

**Comment by izgeri** [comment]
Note: we were able to reproduce the problem locally with the v0.19.1 release. We deployed authenticator client v0.19.1 as usual and modified the deployment (via 'kubectl edit ...') to add in this security context:
```
securityContext:
  runAsNonRoot: true
  runAsUser: 65534
  fsGroup: 65534
```
Then the container crashed as described in this issue. Switching the authenticator client to the older 0.16.0 version, the crashes disappeared.

The issue appears to be that the new Authenticator images (versions 0.18.0+) set the permissions on `/etc/conjur/ssl/` directory to `770` and assume an `authenticator` user, so user 65534 can't access files in that directory.

One more data point. In my setup I was able to eliminate the problem by adding this to the Pod
```
securityContext:
  runAsGroup: 0
```
I think this is the key factor. The newer authenticator clients can run as arbitrary (non-root) users, but the user's group has to be set to '0'.
